test/e2e/sched/utils/deployment.go

```diff
@@ -42,3 +45,18 @@ func GetDeploymentByOwnerReference(uid types.UID) (*v1.Deployment, error) {
 	}
 	return nil, fmt.Errorf("failed to get deployment with uid: %s", uid)
 }
+
+func ListPodsByDeployment(aclient client.Client, deployment *v1.Deployment) ([]corev1.Pod, error) {
```

Comment:
not a fan of passing pointers to data which the function is not expected to mutate, like in this case. We should let the compiler figure out, and force its hands only iff we know this turns out to be a bottleneck (unlikely).